Packs/ProofpointEmailSecurity/Integrations/ProofpointEmailSecurityEventCollector/ProofpointEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,130 @@
+from contextlib import contextmanager
+from enum import Enum
+from CommonServerPython import *  # noqa: F401
+from websockets.sync.client import connect
+from websockets.sync.connection import Connection
+from dateutil import tz
+import traceback
+
+VENDOR = "proofpoint"
+PRODUCT = "email_security"
+
+URL = "wss://{host}/v1/stream?cid={cluster_id}&type={type}&sinceTime={time}"
+
+
+FETCH_INTERVAL_IN_SECONDS = 60
+TIMEOUT = 10
+
+
+class EventType(str, Enum):
+    MESSAGE = "message"
+    MAILLOG = "maillog"
+
+
+def is_interval_passed(fetch_start_time: datetime, fetch_interval: int) -> bool:
+    return datetime.utcnow() > fetch_start_time + timedelta(seconds=fetch_interval)
+
+
+@contextmanager
+def websocket_connections(
+    host: str, cluster_id: str, api_key: str, since_time: str | None = None, to_time: str | None = None
+):
+    url = URL
+    if not since_time:
+        since_time = datetime.utcnow().isoformat()
+    if to_time:
+        url += f"&toTime={to_time}"
+    demisto.info("Starting websocket connection")
+    extra_headers = {"Authorization": f"Bearer {api_key}"}
+    with connect(
+        url.format(host=host, cluster_id=cluster_id, type=EventType.MESSAGE.value, time=since_time),
+        additional_headers=extra_headers,
+    ) as message_connection, connect(
+        url.format(host=host, cluster_id=cluster_id, type=EventType.MAILLOG.value, time=since_time),
+        additional_headers=extra_headers,
+    ) as maillog_connection:
+        yield message_connection, maillog_connection
+
+
+def fetch_events(event_type: EventType, connection: Connection, fetch_interval: int) -> list[dict]:
+    """
+    This function fetches events from the websocket connection for the given event type, for the given fetch interval
+
+    Args:
+        event_type (EventType): The event type to fetch (MAILLOG, MESSAGE)
+        connection (Connection): the websocket connection to the event type
+        fetch_interval (int): the interval of events to fetch, in seconds
+
+    Returns:
+        list[dict]: A list of events
+    """
+    events: list[dict] = []
+    fetch_start_time = datetime.utcnow()
+    while not is_interval_passed(fetch_start_time, fetch_interval):
+        try:
+            event = json.loads(connection.recv(timeout=TIMEOUT))
```

Comment:
Get the event_id once, and use it in all the places. No need to repeat this.
```suggestion
            event = json.loads(connection.recv(timeout=TIMEOUT))
            event_id = event.get('id', event.get('guid'))
```